test/test_StateMachine.py

```diff
@@ -367,3 +367,34 @@ def test_reset(self):
         self.assertEqual(sm.state, 'bar')
         sm.reset()
         self.assertIsNone(sm.state)
+
+    def test_can(self):
+        sm = StateMachine({
+            'initial': 'init',
+            'transitions': {
+                ('init', 'foo'): lambda: True,
+                ('foo', 'bar'): lambda: True,
+                ('bar', 'foo'): lambda: True,
+                ('bar', 'init'): lambda: True
+            }
+        })
+
+        self.assertEqual(sm.state, None)
+        self.assertIs(sm.can('init'), True)
+        self.assertIs(sm.can('foo'), False)
+        self.assertIs(sm.can('bar'), False)
+
+        sm.process(0)
+        self.assertEqual(sm.state, 'init')
+        self.assertIs(sm.can('foo'), True)
+        self.assertIs(sm.can('bar'), False)
```

Comment:
Ah you're right, I forgot to test these. But since it's in principle possible to have 'self-transitions' and `on_exit` and `on_entry` may perform meaningful work even for such a transition I think I'd prefer adding this scenario instead of "forbidding it".